pkg/core/http/cookie_test.go

```diff
@@ -103,3 +103,32 @@ func TestUpdateCookie(t *testing.T) {
 		})
 	})
 }
+
+func TestCookieDomainFromETLDPlusOneWithoutPort(t *testing.T) {
+	Convey("CookieDomainFromETLDPlusOneWithoutPort", t, func() {
+		check := func(in string, out string) {
+			actual := CookieDomainFromETLDPlusOneWithoutPort(in)
+			So(out, ShouldEqual, actual)
+		}
+		check("localhost", "")
```

Comment:
I think this is a bug? It should be localhost. If we cannot derive eTLD+1, we may want to return the domain directly, since `localhost` is a TLD.

Reply:
It is an issue of golang publicsuffix :(
However, returning empty string is part of the contract. If eTLD+1 cannot be derived, returning empty string means use host-only cookie.